### ISSUE TITLE
[E0308] Mismatched types error not detected

### ISSUE THREAD

**Issue by andreisilviudragnea** [intellij-rust/intellij-rust#1661]
There should be an [E0308](https://doc.rust-lang.org/error-index.html#E0308) detected at line `3` in the following snippet, but there is not. The plugin version tested is `0.1.0.2065`

```rust
fn main() {
    let mut spaces = "   ";
    spaces = spaces.len();
}
```

**Comment by vlad20012** [comment]
Currently we do not do type checking inspections. You can enable `use cargo check to analyze code` in settings to see lints from cargo check.
![image](https://user-images.githubusercontent.com/3221931/30210746-6945d994-94a6-11e7-86bc-342557b13323.png)
